OpenRA.Mods.Cnc/Activities/LayMines.cs

```diff
@@ -66,10 +66,14 @@ public override bool Tick(Actor self)
 			if (IsCanceling)
 				return true;
 
-			if ((minefield == null || minefield.Contains(self.Location)) && CanLayMine(self, self.Location))
+			if (minefield == null || minefield.Contains(self.Location))
 			{
-				if (rearmableInfo != null && ammoPools.Any(p => p.Info.Name == minelayer.Info.AmmoPoolName && !p.HasAmmo))
+				// If we don't have enough ammo to lay a mine
+				if (!ammoPools.Any(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage))
```

Comment:
```suggestion
				if (ammoPools.Length != 0 && !ammoPools.Any(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage))
```

We also don't want to rearm if have no ammo pools